examples/cloud-run/main.tf

```
@@ -0,0 +1,162 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  key_name = "${var.key}-${random_id.random_topic_id_suffix.hex}"
+}
+
+resource "random_id" "random_topic_id_suffix" {
+  byte_length = 2
+}
+
+resource "google_pubsub_topic" "secret_topic" {
+  project = var.project_id
+  name    = "topic-${random_id.random_topic_id_suffix.hex}"
+}
+
+resource "google_project_service_identity" "secretmanager_identity" {
+  provider = google-beta
+  project  = var.project_id
+  service  = "secretmanager.googleapis.com"
+}
+
+resource "google_pubsub_topic_iam_member" "sm_sa_publisher" {
+  project = var.project_id
+  role    = "roles/pubsub.publisher"
+  member  = "serviceAccount:${google_project_service_identity.secretmanager_identity.email}"
+  topic   = google_pubsub_topic.secret_topic.name
+}
+
+resource "google_storage_bucket" "bucket" {
+  project                     = var.project_id
+  name                        = "${var.project_id}-pub-sub-source" # Every bucket name must be globally unique
+  location                    = var.region
+  uniform_bucket_level_access = true
+}
+
+data "archive_file" "src" {
+  type        = "zip"
+  source_dir  = "./cloudrun"
+  output_path = "cloudrun.zip"
+}
```

Consider more explicit names, like:

```suggestion
data "archive_file" "compress_cloudrun_sourcecode" {
  type        = "zip"
  source_dir  = "./cloudrun_sourcecode"
  output_path = "cloudrun_sourcecode.zip"
}
```